hq_superset/oauth.py

```diff
@@ -33,6 +36,53 @@ def set_oauth_session(self, provider, oauth_response):
         # }
         session["oauth_response"] = oauth_response
 
+    def ensure_domain_role_created(self, domain):
+        # This inbuilt method creates only if the role doesn't exist.
+        return self.add_role(get_role_name_for_domain(domain))
+
+    def ensure_schema_perm_created(self, domain):
+        menu_name = self.get_schema_perm(get_ucr_database(), domain)
+        permission = self.find_permission_view_menu("schema_access", menu_name)
+        if not permission:
+            permission = self.add_permission_view_menu("schema_access", menu_name)
+        return permission
+
+    def sync_domain_role(self, domain):
+        # This creates DB schema, role and schema permissions for the domain and
+        #   assigns the role to the current_user
+        # self.create_schema(domain)
+        permission = self.ensure_schema_perm_created(domain)
+        role = self.ensure_domain_role_created(domain)
+        role.permissions = [permission]
+        # Todo; may need to add the permission to admin role as well
+        self.get_session.add(role)
+        self.get_session.commit()
+        import pdb; pdb.set_trace()
+        # Filter out other domain roles
+        filtered_roles = [
+            r
+            for r in current_user.roles
+            if not r.name.startswith(DOMAIN_ROLE_PREFIX)
+        ]
+        # Add the domain's role
+        current_user.roles = filtered_roles + [role]
+        self.get_session.add(current_user)
+        self.get_session.commit()
+
+
+def get_ucr_database():
+    from superset import db
+    from superset.models.core import Database
+    # Todo; get actual DB once that's implemented
+    return db.session.query(Database).filter_by(database_name="HQ Data").one()
```

Comment:
Is there a way we can add this in a config file somewhere? Even if it is something like `settings` for django.

Reply:
Yeah, that's a TODO for future, I think we can add it in the superset config file.